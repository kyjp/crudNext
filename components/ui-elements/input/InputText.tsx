import React, { ComponentPropsWithoutRef, FC } from 'react'

type InputTextType = ComponentPropsWithoutRef<'input'>

const InputText: FC<InputTextType> = (props) => {
    const { className, value, onChange, placeholder } = props
    return (
        <input
            className={`text-base px-4 py-2 bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-3xl focus:ring-blue-500 focus:border-blue-500 block w-full p-2.5${className && ` ${className}`}`}
            type="text"
            value={value}
            onChange={onChange}
            placeholder={placeholder}
        />
    )
}

export default InputText