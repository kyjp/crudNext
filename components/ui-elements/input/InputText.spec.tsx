import { render, screen } from "@testing-library/react"
import InputText from "./InputText"
import userEvent from "@testing-library/user-event"
import '@testing-library/jest-dom'


describe('Test group of InputText component', () => {
    const text = "This is InputText"
    // snapshot test
    test("Snapshot Test > renders InputText component", () => {
        const {asFragment} = render(
            <InputText/>
        )
        expect(asFragment()).toMatchSnapshot()
    })

    test("renders InputText component", async () => {
        render(
            <InputText
                placeholder={text}
            />
        )
        expect(await screen.getByPlaceholderText(text)).toBeInTheDocument()
    })

    test("are the characters you entered correct", async () => {
        render(
            <InputText
                placeholder={text}
            />
        )
        const inputElement: HTMLInputElement = screen.getByPlaceholderText(text)
        await userEvent.type(inputElement, text)
        expect(inputElement.value).toBe(text)
    })

    test("when typed, 'mockFn' is called once", async () => {
        const mockFn = vi.fn()
        render(
            <InputText
                placeholder={text}
                defaultValue={""}
                onChange={() => mockFn()}
            />
        )
        const inputElement: HTMLInputElement = screen.getByPlaceholderText(text)
        await userEvent.type(inputElement, "a")
        expect(mockFn).toBeCalled()
        expect(mockFn).toBeCalledTimes(1)
    })
})