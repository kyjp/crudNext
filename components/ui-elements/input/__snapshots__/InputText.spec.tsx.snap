// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test group of InputText component > Snapshot Test > renders InputText component 1`] = `
<DocumentFragment>
  <input
    class="text-base px-4 py-2 bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-3xl focus:ring-blue-500 focus:border-blue-500 block w-full p-2.5undefined"
    type="text"
  />
</DocumentFragment>
`;
